controllers/managedocs_controller.go

```diff
@@ -1058,6 +1079,95 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) reconcilePrometheusRemoteWrite() (*promv1.RemoteWriteSpec, error) {
+	rhobsSecret := &corev1.Secret{}
+	rhobsSecret.Name = prometheusRemoteWriteSecretName
+	rhobsSecret.Namespace = r.namespace
+	if err := r.get(rhobsSecret); err != nil {
+		return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+	}
+	rhobsSecretData := rhobsSecret.Data
+	rhobsSecretID := string(rhobsSecretData[prometheusRemoteWriteIDKey])
+	if rhobsSecretID == "" {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", prometheusRemoteWriteIDKey)
+	}
+	rhobsSecretKey := string(rhobsSecretData[prometheusRemoteWriteSecretKey])
+	if rhobsSecretKey == "" {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", prometheusRemoteWriteSecretKey)
+	}
+
+	rhobsConfigMap := &corev1.ConfigMap{}
+	rhobsConfigMap.Name = promRemoteWriteRelabelConfigMapName
+	rhobsConfigMap.Namespace = r.namespace
```

Comment:
- maybe this initilization should go into `initReconciler`
- we can access `DeploymentType` in there as well

Reply:
Same as previous comment. Let's wait for others verdict on this

Reply:
changed implementation